src/ranged.cpp

```diff
@@ -831,8 +834,11 @@ dispersion_sources calculate_dispersion( const map &m, const Character &who, con
 
 static int calc_gun_volume( const item &gun )
 {
-    int noise = gun.type->gun->loudness;
-    for( const auto mod : gun.gunmods() ) {
+    // Inherit suppressor modifiers if relevant (e.g. KSG second mag) but still use current ammo
+    const item &parent = gun.parent_item() != nullptr &&
+                         gun.has_flag( flag_USE_PARENT_GUN ) ? *gun.parent_item() : gun;
```

Comment:
```suggestion
    const item &parent = ( gun.parent_item() != nullptr &&
                         gun.has_flag( flag_USE_PARENT_GUN ) ) ? *gun.parent_item() : gun;
```
same suggestion as earlier.